os/fs/smartfs/smartfs_smart.c

```diff
@@ -178,6 +178,9 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 	uint16_t parentdirsector;
 	const char *filename;
 	struct smartfs_ofile_s *sf;
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+	struct smartfs_entry_s invalidentry;
```

Comment:
Actually target number of sector where invalid entry saved, and offset is enough to use.

Reply:
Using `smartfs_entry_s invalidsector` in smartfs_open() allows using smartfs_replaceentry() function along the same lines as for smartfs_rename() without the need for adding two additional parameters for sector no. and offset.